tableauserverclient/server/server.py

```diff
@@ -47,6 +58,29 @@ def _set_auth(self, site_id, user_id, auth_token):
         self._user_id = user_id
         self._auth_token = auth_token
 
+    def _get_legacy_version(self):
+        response = self._session.get(self.server_address + "/auth?format=xml")
+        info_xml = ET.fromstring(response.content)
+        prod_version = info_xml.find('.//product_version').text
+        version = _PRODUCT_TO_REST_VERSION.get(prod_version, '2.1')  # 2.1
+        return version
+
+    def _determine_highest_version(self):
+        try:
+            old_version = self.version
+            self.version = "2.4"
+            version = self.server_info.get().rest_api_version
+        except ServerInfoEndpointNotFoundError:
+            version = self._get_legacy_version()
+
+        finally:
```

Comment:
This seems a bit weird.  We are setting the version, determining the new version, and then resetting the old version.  I think with the way this is written, we probably don't need _determine_highest_version to be it's own function.  In this case, we could do something like the following pseudo-code

```
save old version
set version to 2.4
Try to call get rest api version
If exception, try to get legacy version
If no legacy version found, restore old value.
```

I'm not sure which version would be clearer to understand.